frontend/packages/multicluster-sdk/src/api/use-fleet-k8s-watch-resource/use-fleet-k8s-watch-resource.ts

```diff
@@ -30,17 +47,35 @@ const getCacheKey = ({
   return [cluster, model?.apiVersion, model?.kind, namespace, name].join('|')
 }
 
+/**
+ * A hook for watching Kubernetes resources, with support for watching resources on remote clusters through the fleet API.
+ *
+ * This hook behaves as a wrapper around the OpenShift Console's `useK8sWatchResource` hook.
+ * When watching a resource on a remote cluster (i.e., `cluster` is specified in `initResource` and is not the hub cluster),
+ * this hook will use the fleet API to fetch and watch the resource. It maintains its own cache (`fleetResourceCache`)
+ * and WebSocket connections (`fleetSocketCache`) for this purpose.
+ *
+ * When watching a resource on the hub cluster or if the fleet API is not available, it falls back to using `useK8sWatchResource`.
+ *
+ * @template R - The type of the Kubernetes resource.
+ * @param hubClusterName - The name of the hub cluster.
+ * @param initResource - The resource to watch. Can be null to disable the watch.
+ * @returns A tuple containing the watched resource data, a boolean indicating if the data is loaded, and any error that occurred.
+ *          If the watch is for a remote cluster, it uses a custom implementation. Otherwise, it returns the result of `useK8sWatchResource`.
+ */
```

Comment:
This is an internal function. It's the other one (without the hubClusterName param) that we want documented. I would avoid talking about implementation details in the docs and instead talk about how to use the function and its behaviour. So you can say that it uses websockets and that they are cached/pooled, but you don't need to talk about where in the code that is done. Also, I'm not sure "fleet API" is clear... I would think this function is part of the "fleet API"... so a bit circular to say it uses the fleet API or works through the fleet API.